src/app/(proper_react)/redesign/(public)/Faq.tsx

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, useRef, useState } from "react";
+import { useL10n } from "../../../hooks/l10n";
+import styles from "./LandingView.module.scss";
+import { CloseBigIcon } from "../../../components/server/Icons";
+import { useTelemetry } from "../../../hooks/useTelemetry";
+import { useButton, useFocusRing } from "react-aria";
+
+export type FaqItemProps = {
+  question: string;
+  answer: string | ReactNode;
+  isExpanded: boolean;
+  id: string;
+  onExpandAnswer: () => void;
+};
+
+const FaqItem = (props: FaqItemProps) => {
+  const l10n = useL10n();
+  const buttonRef = useRef(null);
+
+  const { buttonProps } = useButton(
+    {
+      onPress: props.onExpandAnswer,
+    },
+    buttonRef,
+  );
+  const { focusProps } = useFocusRing();
+
+  return (
+    <>
+      <dt
+        {...buttonProps}
+        {...focusProps}
+        tabIndex={0}
+        className={styles.faqQuestion}
+        aria-expanded={props.isExpanded}
+        aria-controls={`faq-${props.id}`}
```

Comment:
I believe this needs to point to another element (in this case, the related `<dd>`) that has the given value as its `id`. If you want to avoid needing to come up with IDs yourself, you can also use React's [`useId`](https://react.dev/reference/react/useId) hook.

Reply:
I think both are true: you need `aria-expanded` to indicate whether the item that is controlled by _this_ item is visible or not, and `aria-controls` to indicate which item that is. So to adapt an example from MDN:

```html
<button
  aria-expanded="false"
  aria-controls="the-desc">
  Expand
</button>
<p id="the-desc" hidden>
  This content is current hidden
</p>
```

So the `<p>` has an ID, and the `<button>` has both `aria-expanded`, and an `aria-controls` that points to the `<p>`'s ID.